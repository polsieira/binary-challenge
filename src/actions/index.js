export const checkIsLoading = (bool) => ({
  type: 'IS_LOADING',
  isLoading: bool
})

export const hasErrored = (message) => ({
  type: 'HAS_ERRORED',
  message
})

export const addExoplanets = (exoplanets) => ({
  type: 'ADD_EXOPLANETS',
  exoplanets
})

export const loginUser = (userInfo) => ({
  type: 'LOGIN_USER',
  userInfo
})

export const addFilters = (filters) => ({
  type: 'ADD_FILTERS',
  filters
})