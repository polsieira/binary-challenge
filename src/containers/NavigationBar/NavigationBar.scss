.NavigationBar {
  align-items: center;
  background: white;
  display: flex;
  height: 75px;
  justify-content: space-between;
  width: 100%;

  .logo {
    height: 75px;
    margin-left: 20px;
  }

  a {
    text-decoration: none;
  }

  .menu-button {
    height: 45px;
    margin-right: 20px;
    width: 100px;
  }
}