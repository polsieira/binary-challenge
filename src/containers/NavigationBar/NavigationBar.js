import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import Button from '@material-ui/core/Button';
import Menu from '@material-ui/core/Menu';
import MenuItem from '@material-ui/core/MenuItem';
import Fade from '@material-ui/core/Fade';
import './NavigationBar.scss'
import { bindActionCreators } from 'redux';
import PropTypes from 'prop-types';
import { loginUser } from '../../actions';
import logo from '../../assets/EXOGO.svg';

export const NavigationBar = ({ user, loginUser }) => {

  const [anchorEl, setAnchorEl] = React.useState(null);
  const open = Boolean(anchorEl);

  const handleClick = event => {
    setAnchorEl(event.currentTarget);
  };

  const handleClose = (e) => {
    setAnchorEl(null);
    if (e.target.id === 'logout') {
      loginUser({
        name: '',
        id: null,
        isLoggedIn: false
      })
    }
  };

  return (
    <nav className="NavigationBar">
      <img className="logo" alt="exogo logo" src={logo} />
      {user.isLoggedIn ?
        <>
          <Button className="menu-button" aria-controls="fade-menu" aria-haspopup="true" onClick={handleClick}>
            Profile
          </Button>
          <Menu
            id="fade-menu"
            anchorEl={anchorEl}
            keepMounted
            open={open}
            onClose={handleClose}
            TransitionComponent={Fade}
          >
            <MenuItem id='saved-exoplanets' onClick={handleClose}>Saved Exoplanets</MenuItem>
            <MenuItem id='logout' onClick={handleClose}>Logout</MenuItem>
          </Menu>
        </> :
        <Link to='/login' >
          <Button className="menu-button" onClick={handleClick}>
            Login
          </Button>
        </Link>}
    </nav>
  )
}

export const mapStateToProps = ({ user }) => ({
  user
})

export const mapDispatchToProps = dispatch => {
  return bindActionCreators({
    loginUser,
  }, dispatch);
};

export default connect(mapStateToProps, mapDispatchToProps)(NavigationBar);

NavigationBar.propTypes = {
  user: PropTypes.object,
}