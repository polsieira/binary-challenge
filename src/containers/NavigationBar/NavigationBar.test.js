import React from 'react';
import { shallow } from 'enzyme';
import { NavigationBar, mapStateToProps, mapDispatchToProps } from './NavigationBar';
import { loginUser } from '../../actions';

describe('NavigationBar', () => {
  describe('NavigationBar component', () => {
    let wrapper;
    const mockLoginUser = jest.fn();
    const mockUser = {
      name: 'Meep Meep',
      id: 5,
      isLoggedIn: true
    }

    beforeEach(() => {
      wrapper = shallow(<NavigationBar user={mockUser} loginUser={mockLoginUser} />);
    });

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should match the snapshot when there is no user', () => {
      wrapper = shallow(<NavigationBar user={({ isLoggedIn: false })} loginUser={mockLoginUser} />);
      expect(wrapper).toMatchSnapshot();
    })

    it.skip('should call setAnchorEl when handleClick is invoked', () => {
      const mockEvent = {
        currentTarget: 'Target'
      };

      wrapper.instance().handleClick(mockEvent);

      expect(wrapper.setAnchorEl).toHaveBeenCalledWith(mockEvent.currentTarget)
    });

  });

  describe('mapStateToProps', () => {
    it('should return an object with a movies array, an error message and isLoading property', () => {
      const mockState = {
        user: {
          name: 'Meep meep',
          id: 5,
          isLoggedIn: true,
        },
        mockVariable: 'Mock mock mock'
      };

      const expected = {
        user: {
          name: 'Meep meep',
          id: 5,
          isLoggedIn: true,
        }
      };

      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps).toEqual(expected);
    });
  });

  describe('mapDispatchToProps', () => {

    const mockDispatch = jest.fn();


  });

});
