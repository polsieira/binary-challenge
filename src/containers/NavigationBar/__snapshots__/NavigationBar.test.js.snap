// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationBar NavigationBar component should match the snapshot 1`] = `
<nav
  className="NavigationBar"
>
  <img
    alt="exogo logo"
    className="logo"
    src="EXOGO.svg"
  />
  <WithStyles(ForwardRef(Button))
    aria-controls="fade-menu"
    aria-haspopup="true"
    className="menu-button"
    onClick={[Function]}
  >
    Profile
  </WithStyles(ForwardRef(Button))>
  <WithStyles(ForwardRef(Menu))
    TransitionComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "propTypes": Object {
          "children": [Function],
          "in": [Function],
          "onEnter": [Function],
          "onExit": [Function],
          "style": [Function],
          "timeout": [Function],
        },
        "render": [Function],
      }
    }
    anchorEl={null}
    id="fade-menu"
    keepMounted={true}
    onClose={[Function]}
    open={false}
  >
    <WithStyles(ForwardRef(MenuItem))
      id="saved-exoplanets"
      onClick={[Function]}
    >
      Saved Exoplanets
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      id="logout"
      onClick={[Function]}
    >
      Logout
    </WithStyles(ForwardRef(MenuItem))>
  </WithStyles(ForwardRef(Menu))>
</nav>
`;

exports[`NavigationBar NavigationBar component should match the snapshot when there is no user 1`] = `
<nav
  className="NavigationBar"
>
  <img
    alt="exogo logo"
    className="logo"
    src="EXOGO.svg"
  />
  <Link
    to="/login"
  >
    <WithStyles(ForwardRef(Button))
      className="menu-button"
      onClick={[Function]}
    >
      Login
    </WithStyles(ForwardRef(Button))>
  </Link>
</nav>
`;
